pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1148,18 +1146,32 @@ func buildDefaultCluster(env *model.Environment, name string, discoveryType apiv
 		}
 	}
 
-	defaultTrafficPolicy := buildDefaultTrafficPolicy(env, discoveryType)
+	defaultTrafficPolicy := buildDefaultTrafficPolicy(env, discoveryType, direction, port)
 	applyTrafficPolicy(env, cluster, defaultTrafficPolicy, nil, nil, "",
 		DefaultClusterMode, direction)
 	return cluster
 }
 
-func buildDefaultTrafficPolicy(env *model.Environment, discoveryType apiv2.Cluster_DiscoveryType) *networking.TrafficPolicy {
+func buildDefaultTrafficPolicy(env *model.Environment, discoveryType apiv2.Cluster_DiscoveryType,
+	direction model.TrafficDirection, port *model.Port) *networking.TrafficPolicy {
 	lbPolicy := DefaultLbType
 	if discoveryType == apiv2.Cluster_ORIGINAL_DST {
 		lbPolicy = networking.LoadBalancerSettings_PASSTHROUGH
 	}
+
+	// Set default circuit breaking for Envoy graceful shutdown
+	var outlierDetectionPolicy *networking.OutlierDetection
+	if direction == model.TrafficDirectionOutbound && port.Protocol.IsHTTP() {
+		outlierDetectionPolicy = &networking.OutlierDetection{
```

Comment:
Couple of points here

- While passive health check (outlier detection) is a good setting to have, it is not required for graceful shutdown cases. Once active health checks are enabled and you invoke `/healthcheck/fail` via admin end point, the host would be considered unhealthy/draining and would be removed from the LB pool.
- If you still enable passive health checks for some other reason, I think defaulting 10 to `MaxEjectionPercent` would have issues especially for smaller clusters and may lead to ejection overflows

Reply:
Could you give an example of ejection overflow issues?

Reply:
If you have 3 nodes in the cluster, with 10% ejection percent, at any time you would be able to eject only one instance even though more than one instance may be behaving badly.

Reply:
This is a good point, are there any guidelines on sensible default ejection percentages?

Reply:
This doc I created in Envoy might help https://www.envoyproxy.io/docs/envoy/latest/faq/transient_failures. We also set to 50% and it seems to help with ejection overflows.

Reply:
But if active health checking is not enabled, what is the purpose of calling `/healthcheck/fail` end point during shutdown. IIUC, when you call `/healthcheck/fail` - it will fail any inbound health checks (configured via upstream clusters active health check configuation). But if you do not enable that, who will use this health check failure?

Reply:
`healthcheck/fail` also 503s all new requests to any (HTTP) listeners with the hc filter. This triggers the outlier detection. This PR effectively puts Envoy into quiesce mode when `pilot-agent` is sent SIGTERM. In this mode, it will reject any new inbound requests but allow new outbound connections.

This enables a couple of benefits:
- Ongoing connections have a period of time to complete and are not severed until the Envoy process is killed (30 seconds by default here).
- Some applications when shutting down need to make outbound connections (see linked issue #7136).

Reply:
I manually tested the `healthcheck/fail` behaviour using [this playground](https://github.com/liamawhite/envoy-playground).

Reply:
>> healthcheck/fail also 503s all new requests to any (HTTP) listeners with the hc filter

I did not know about this. This is good to know. I thought it will only respond to health checks. If it responds to *any* http request, it will be useful.

Reply:
It is not documented, or at least not documented clearly. I had to lean on a couple of Envoy maintainers to work out how to do this 😂.

I think it's that the passthrough mode on all request (with no header filter) basically just enables a switch where you can just switch off the passing through of requests. This is my mental model from discussing with some of the Envoy team, the code may actually do something different though.